guppylang/cfg/analysis.py

```diff
@@ -85,83 +85,92 @@ def run(self, bbs: Iterable[BB]) -> Result[T]:
 
 # For live variable analysis, we also store a BB in which a use occurs as evidence of
 # liveness.
-LivenessDomain = dict[str, BB]
+LivenessDomain = dict[P, BB]
 
 
-class LivenessAnalysis(BackwardAnalysis[LivenessDomain]):
+class LivenessAnalysis(BackwardAnalysis[LivenessDomain[P]], Generic[P]):
     """Live variable analysis pass.
 
     Computes the variables that are live before the execution of each BB. The analysis
     runs over the lattice of mappings from variable names to BBs containing a use.
     """
 
-    def eq(self, live1: LivenessDomain, live2: LivenessDomain) -> bool:
+    stats: dict[BB, VariableStats[P]]
```

Comment:
nit: `BB` now looks a bit like a type variable.... :-(

Reply:
Would you prefer renaming to `BasicBlock`?

Reply:
That would be clearer, but might be a bit long (ymmv). You could also consider `Block`....and I'm sure I've seen just `Blk` although I wouldn't say I'm a fan. Not sure about `BBlk`...